artemis-jms-client/src/main/java/org/apache/activemq/artemis/jms/client/ActiveMQConnectionFactory.java

```diff
@@ -88,20 +83,6 @@
 
    private boolean finalizeChecks;
 
-   @Override
-   public void writeExternal(ObjectOutput out) throws IOException {
-      URI uri = toURI();
-
-      try {
-         out.writeUTF(uri.toASCIIString());
```

Comment:
Wouldn't this will break compatibility.
Think of anyone serializing this class.

Looking at the JNDISTorage (new class) I see that this is calling now writeObject(getProperties());

As opposed to the writeURI.

The point was to write a string to be better compatible between versions.

Can we add back the format we had here?

Reply:
No it is serialising fine, in fact this is what tomcat resources is doing, this is exactly what we are purposely changing to fix the issue, it needs to serialise into an array of StringRefAddr (properties).